reconcile/terraform_resources.py

```diff
@@ -549,27 +549,49 @@ def init_working_dirs(
     return ts, working_dirs
 
 
+def filter_accounts_by_name(
+    accounts: Iterable[Mapping[str, Any]], filter: Iterable[str]
+) -> Collection[Mapping[str, Any]]:
+    return [ac for ac in accounts if ac["name"] in filter]
+
+
+def exclude_accounts_by_name(
+    accounts: Iterable[Mapping[str, Any]], filter: Iterable[str]
+) -> Collection[Mapping[str, Any]]:
+    return [ac for ac in accounts if ac["name"] not in filter]
+
+
+def validate_account_names(
+    accounts: Collection[Mapping[str, Any]], names: Collection[str]
+) -> None:
+    if len(accounts) != len(names):
+        missing_names = set(names) - {a["name"] for a in accounts}
+        raise ValueError(
+            f"Accounts {missing_names} were provided as arguments, but not found in app-interface. Check your input for typos or for missing AWS account definitions."
+        )
+
+
 def setup(
     dry_run: bool,
     print_to_file: str,
     thread_pool_size: int,
     internal: str,
     use_jump_host: bool,
-    account_names: Optional[Collection[str]],
+    include_accounts: Optional[Collection[str]],
+    exclude_accounts: Optional[Collection[str]],
 ) -> tuple[ResourceInventory, OC_Map, Terraform, ExternalResourceSpecInventory]:
     accounts = queries.get_aws_accounts(terraform_state=True)
-    if account_names:
-        accounts = [n for n in accounts if n["name"] in account_names]
-        if len(accounts) != len(account_names):
-            # Some of the passed account names don't exist in app-interface
-            acc_names = tuple(
-                a
-                for a in account_names
-                if a not in tuple(account["name"] for account in accounts)
-            )
-            raise ValueError(
-                f"Accounts {acc_names} were provided as arguments, but not found in app-interface. Check your input for typos or for missing AWS account definitions."
-            )
+    if not include_accounts and exclude_accounts:
```

Comment:
I'll investigate that.

Reply:
I'm failing to see how this things relate to each other.

Reply:
By reading the code, what I think is that Sharding in the runtime context is on higher level of abstraction than the Sharding in terraform-resources. Maybe moving this to there is going to make the code there too specific? 

Also, it seems like Sharding in runtime is more pointed to run without dry-run, while my changes here are specific for dry run mode.

Reply:
I apologize, this is more of the appropriate method: https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/runtime/runner.py#L172 
The main reason I mentioned this was because the new framework is trying take care of the state of the dry-run before the integration's run method gets called, which keep each integration from having to worry about it. 
I just saw Gerd's comment on your ticket, since he is the designer for this new separation of responsibility, I won't insist on this.